src/v/kafka/server/group.cc

```diff
@@ -1305,30 +1473,72 @@ ss::future<heartbeat_response> group::handle_heartbeat(heartbeat_request&& r) {
 ss::future<leave_group_response>
 group::handle_leave_group(leave_group_request&& r) {
     vlog(_ctxlog.trace, "Handling leave group request {}", r);
+    /**
+     * Leave group prior to 3 had were requesting only a single member leave,
+     * add member id to members list to simplify downstream processing
+     */
+    if (r.version < api_version(3)) {
+        auto err = member_leave_group(r.data.member_id, std::nullopt);
+        return make_leave_error(err);
+    }
```

Comment:
I'm not sure I follow the comment and the code.

Should this also check for `in_state(group_state::dead)` below?

Reply:
this comment is obsolete, i will change it, great find